third_party/terraform/resources/resource_compute_attached_disk.go

```diff
@@ -76,10 +76,21 @@ func resourceAttachedDiskCreate(d *schema.ResourceData, meta interface{}) error
 		return err
 	}
 
-	diskName := GetResourceNameFromSelfLink(d.Get("disk").(string))
+	disk := d.Get("disk").(string)
+	diskName := GetResourceNameFromSelfLink(disk)
+	diskSrc := fmt.Sprintf("projects/%s/zones/%s/disks/%s", zv.Project, zv.Zone, diskName)
+
+	// Check if the disk is a regional disk
+	if strings.Contains(disk, "regions") {
+		rv, err := ParseRegionDiskFieldValue(disk, d, config)
+		if err != nil {
+			return err
+		}
+		diskSrc = fmt.Sprintf("projects/%s/regions/%s/disks/%s", rv.Project, rv.Region, diskName)
```

Comment:
can this just be `diskSrc = rv.RelativeLink()`? And the other one above be `zv.RelativeLink`?

Reply:
The second link can be, but the first can't. zv are the zonal fields parsed from the compute instance, not the disk.

Reply:
Ah right, that makes sense.